### PR TITLE
Split APIHelper out into its own file

### PR DESCRIPTION
**Pull request by n-peugnet**
This allow to only import it when needed, speeding up the CLI for easy tasks, especially the [auto-completion provided by click][1] (that I intend to add to the Debian package). This is most noticable when site packages are enabled (as with the command `python -m virtualenv --system-site-packages`) or on global installations, like with the official Debian package.

Before

	$ time synadm > /dev/null
	real	0m0,610s
	user	0m0,537s
	sys	0m0,072s

After

	$ time synadm > /dev/null
	real	0m0,086s
	user	0m0,077s
	sys	0m0,009s

[1]: https://click.palletsprojects.com/en/latest/shell-completion/

References

- https://blog.tomrochette.com/problems/2020/03/07

### CODE REVIEW

**Comment by n-peugnet** [comment]
> Some actual benchmarking

Thank you, I wasn't motivated enough to do it!

> This really looks like filesystem caching in action, so this benchmark likely isn't good.

Not really, I didn't take the first run, subsequent runs are all about the same time.

Also note that it really depends on the quantity of python libraries installed on the system. For instance in a bare-bones virtualenv it is almost not noticeable.

**Comment by n-peugnet** [comment]
> > Also note that it really depends on the quantity of python libraries installed on the system. For instance in a bare-bones virtualenv it is almost not noticeable.
> 
> Hmm, I'll try barebones venv in both cases later.

Sorry I meant the difference is almost not noticeable. Meaning it is almost the same time for both my changes and master.

**Comment by n-peugnet** [comment]
Some more numbers to explain the differences:

```console
$ # master in venv made with `python -m virtualenv --system-site-packages`
$ strace synadm 2>&1 | wc -l
13251
$ # this branche in venv made with `python -m virtualenv --system-site-packages`
$ strace synadm 2>&1 | wc -l
1668
$ # master in normal venv
$ strace synadm 2>&1 | wc -l
4448
$ # this branch in normal venv
$ strace synadm 2>&1 | wc -l
1537
```

**Review by JOJ0 — CHANGES_REQUESTED** [review]
Hi to both of you,
awesome idea. Splitting out into a separate file is a good idea even if it was not for performance. I didn't realize that there is a Debian package and you are the maintainer @n-peugnet, that is great news as well! Thanks so much for taking care of that. Tab completion for synadm also sounds like a very handy feature for the package. We certainly should work towards making that a pleasent user experience.

One change request: Please let's name that file `_helper.py`.

**Comment by n-peugnet** [comment]
> I didn't realize that there is a Debian package and you are the maintainer @n-peugnet

I am not the [official maintainer](https://tracker.debian.org/pkg/synadm) but I intend to send him a patch to add the completion script in the package.

> Tab completion for synadm also sounds like a very handy feature for the package. We certainly should work towards making that a pleasent user experience.

This is exactly when testing this completion script (generated by click) that loads the python project each time that I noticed the startup was quite slow.

> One change request: Please let's name that file `_helper.py`.

I can do that in a few days. Not that much time right now.

**Comment by n-peugnet** [comment]
> would it be fine if i did it?

Yes of course!

**Comment by JOJ0** [comment]
Hi, sorry for the delay. I just wanted to quickly add some refinements to the docstrings of the modules. Are you sure you have maintainer edits enabled?

File: synadm/cli/__init__.py [hunk 1]
```diff
@@ -18,20 +18,8 @@
 """ CLI base functions and settings
```
Comment (JOJ0) [inline synadm/cli/__init__.py:18]:
```suggestion
""" Contains CLI root-level commands; Subcommands are imported at the bottom of
+this file
```

File: synadm/cli/__init__.py [hunk 2]
```diff
@@ -18,20 +18,8 @@
...
 """
```
[the remark this answers is not in the document]
Comment (n-peugnet) [reply synadm/cli/__init__.py:19]: > Are you sure you have maintainer edits enabled?

I'm really sorry I didn't find how to check or change this. But I thought it was on by default.

About this suggestion, I think you meant this:

```suggestion
""" CLI root-level commands; Subcommands are imported at the bottom of file
"""
```

[the remark this answers is not in the document]
Comment (JOJ0) [reply synadm/cli/__init__.py]: No problem at all :-)

**Pull request merged**